### PR TITLE
[fix](nereids)push filter through window, using slot equal-set

### PR DESCRIPTION
**Pull request by englefly**
## Proposed changes
example:
```
filter (y=1)
+-- window( ... partition by x)
    +-- project( A as x, A as y)
```

filter(y=1) is equivalent to filter(x=1),
because x and y are in the same equal-set in window#logicalProperties.
And hence we could push filter(y=1) through window operator

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by englefly** [comment]
run buildall

**Comment by englefly** [comment]
run buildall

**Review by xzj7019 — APPROVED** [review]

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/algebra/Window.java [hunk 1]
```diff
@@ -115,7 +117,19 @@ default Set<SlotReference> getCommonPartitionKeyFromWindowExpressions() {
         int winExprCount = getWindowExpressions().size();
         for (Map.Entry<Expression, Integer> entry : partitionKeyCount.entrySet()) {
             if (entry.getValue() == winExprCount && entry.getKey() instanceof SlotReference) {
-                commonPartitionKeySet.add((SlotReference) entry.getKey());
+                SlotReference slot = (SlotReference) entry.getKey();
```
Comment (xzj7019) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/algebra/Window.java:120]: would better to add that ut in to track.

**Comment by morrySnow** [comment]
add regression test and description

**Review by starocean999 — DISMISSED** [review]

**Review by LiBinfeng-01 — APPROVED** [review]

**Comment by englefly** [comment]
run buildall

**Comment by englefly** [comment]
run buildall

**Review by starocean999 — APPROVED** [review]

**Review by xzj7019 — APPROVED** [review]

**Pull request merged**